### DISCUSSION TITLE
Request Body and  Response Body

### DISCUSSION THREAD

**Discussion by WildCLown** [postmanlabs/postman-flows#266] [Q&A] [answered]
Hey ya, I've been testing out flows in order to ease testing through my aplication, and I don't realy know if the way I'm currently doing things is the optimal one:

1 - When using Send Request I use Assign values to Variable in order to change my {{request_body}}
Is there a simple way to do so? 

![image](https://user-images.githubusercontent.com/32888444/156606739-7b0ffc3e-f5e4-45d4-9796-4bcf1fea9d9e.png)

2 - To check out my responses through the whole flow, I'm printing through tests by using the following script:
pm.test(JSON.stringify(pm.response.json()), function(){
    pm.response.to.have.status(200)
});
Any simple way to make it clean? In my opinion, it's kind weird seeing results through test summary in order to check an particular data (Response body), but yet, haven't found an better way.
![image](https://user-images.githubusercontent.com/32888444/156611496-15a3c2bd-be76-4b73-aeec-9d03ba7cd152.png)

I've just started using it, so I apologize in advance if was posted somewhre else, or if I just didn't find it by an lack of attention 😂

**Comment by samay-v** [comment, marked as answer]
Hey @WildCLown 
This is just the place for questions like this 😄 

1. If I understand correctly, You're trying to assign an object to your body. You can use the block "Create Data" to do this better. 
You could create your object in the block like so: 
![Screenshot 2022-03-04 at 2 56 52 PM](https://user-images.githubusercontent.com/28357106/156736668-d82848d8-b855-4324-b06a-281eb7b4c7d1.png)
Then pass that to the request as input 
![Screenshot 2022-03-04 at 2 58 10 PM](https://user-images.githubusercontent.com/28357106/156736875-9a3b2ada-561f-442c-b3bf-1a5e4c4454d6.png)
Then assign the input to your variable and use the `to-json` operator on it
![Screenshot 2022-03-04 at 2 59 42 PM](https://user-images.githubusercontent.com/28357106/156737292-8da1981b-392d-419b-a51a-5454d0bf8f8e.png)
2. Instead of using the test script, you can attach the terminal block directly to the response output port to see the response data in the terminal window in the context bar on the right
![Screenshot 2022-03-04 at 3 05 23 PM](https://user-images.githubusercontent.com/28357106/156738189-aa969986-104a-4d6d-93d5-d52e2ac06871.png)

**Reply by WildCLown** [reply]
Thanks a lot @samay-v for the instructions
About the Data structure, I think that I'll keep using the raw text as I was using before for now, creating the data structure is fine when you are doing it from scratch, but I have some samples in my collection, it would be really nice if we could have the same view if we want to change the body content

and about the Terminal one, I was trying to get only my response body content, as in terminal, it's shown the whole request message, also, just like it's shown in collection.

Just to make things clear and show my purpose, I'm using flows as an way to test my application xD
so basically I have many of the same request, but with different bodys, excluding or adding some field, kind require lot of Ctrl + C | Ctrl + V...

Yet, your answer helped me a lot to solve another problem I had
